backend/tests/integration-async/conftest.py

```diff
@@ -98,138 +68,25 @@ def celery_app_with_context(app, celery_session_app):
-                except Exception as e:
-                    _last_error = e
-                    break
-                if _expected_result:
-                    if (
-                        not _expected_key
-                        and isinstance(_expected_result, dict)
-                        and "expected_key" in _expected_result
-                        and "expected_result" in _expected_result
-                    ):
-                        _expected_key = _expected_result["expected_key"]
-                        _expected_result = _expected_result["expected_result"]
-                    if _expected_key:
-                        result = get_value(_current_result, _expected_key)
-                    else:
-                        result = _current_result
-
-                    try:
-                        assert result == _expected_result
-                        finished = True
-                    except AssertionError as e:
-                        _last_error = e
-                        finished = False
-                        time.sleep(0.3)
-                        break
-        return _current_result
-
-    return _check
-
-
-@pytest.fixture(scope="class")
-def app_cli(app):
+def cli(inspire_async_app):
     """Click CLI runner inside the Flask application."""
     runner = CliRunner()
-    obj = ScriptInfo(create_app=lambda info: app)
+    obj = ScriptInfo(create_app=lambda info: inspire_async_app)
     runner._invoke = runner.invoke
-    runner.invoke = partial(runner.invoke, obj=obj)
-    return runner
+    runner.invoke = partial(runner._invoke, inspire_cli, obj=obj)
+    yield runner
 
 
 @pytest.fixture(scope="function")
-def generate_records():
-    def _generate(
-        count=10, record_type=LiteratureRecord, data={}, skip_validation=False
-    ):
-        for i in range(count):
-            data = faker.record(
-                record_type.pid_type, data=data, skip_validation=skip_validation
-            )
-            rec = record_type.create(data)
-        db.session.commit()
-
-    return _generate
-
-
-@pytest.fixture(scope="function")
-def create_record():
-    def _create_record(record_type, data=None, skip_validation=False):
-        data = faker.record(
-            record_type,
-            data=data,
-            with_control_number=True,
-            skip_validation=skip_validation,
-        )
-        record = InspireRecord.create(data)
-        db.session.commit()
-        return record
-
-    return _create_record
+def redis(inspire_async_app):
+    redis_url = inspire_async_app.config.get("CACHE_REDIS_URL")
+    redis = StrictRedis.from_url(redis_url, decode_responses=True)
+    redis.flushall()
+    yield redis
+    redis.flushall()
+    redis.close()
 
 
 @pytest.fixture(scope="function")
-def cache(app):
-    redis_client = StrictRedis.from_url(app.config["CACHE_REDIS_URL"])
-    yield redis_client
-    redis_client.flushall()
+def inspire_async_app(app, cache, clear_environment):
```

Comment:
can we just call it ``inspire_app``? just to have the same for all cases

Reply:
:+1: